### ISSUE TITLE
TV Show metadata extraction fails when using nested paths

### ISSUE THREAD

**Issue by flux-inductor** [jellyfin/jellyfin#13297]
### This issue respects the following points:

- [x] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [x] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [x] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [x] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [x] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

The following issue persists in Jellyfin 10.10.3. It was present in 10.9.x as well.

For a few months I have been struggling with Jellyfin being unable to locate metadata for TV shows and defaulting everything to some random Spanish-language TV show. (Previously it was a French-language show, and I'm not sure why it changed after library rebuilds.) Movies have never been a problem. All indications from searching online pointed to a file naming issue, but this didn't make sense as the convention was already reasonably followed, and other software - such as InFuse - had no issues indexing all the content. Adding the year to the titles did not help. Log files were not providing any helpful clues.

Background: I have content on multiple file servers. These are SMB mounted on the Linux host, and the root of the mounts is passed into the Jellyfin Docker container as /media, which results in the example paths as follows:

/media/share1/Movies/etc
/media/share1/TV Shows/Showname 1/Season 1/Showname S01E01.mkv
/media/share2/TV Shows/etc
/media/share3/TV Shows/etc

The "Movies" and "TV Shows" folders were explicitly chosen as the content folders in the web UI when building the libraries. Movies and Shows were placed in separate libraries.

Movies always indexed correctly, but all TV shows indexed as the Spanish-language show "Midnight Family". (Previously they were indexing as a French-Canadian title, but I can't remember the name.) These shows *do not exist* on the file shares, but there would be over a dozen separate instances placed in the library, all pointing to various unrelated content. Logs only showed that "Unknown Season" of this show was added to the library, but didn't explain why.

On a whim, I tried truncating one level of the path by adjusting the mappings into the Docker container as follows:

/media/shareX/ on the host were individually mapped to /mediaX, resulting in the following paths in the container:

/media1/Movies/etc
/media1/TV Shows/Showname 1/Season 1/Showname S01E01.mkv
/media2/TV Shows/etc
/media3/TV Shows/etc

Once I did this, and added these folders to a library, the metadata was located and the shows indexed correctly.

### Reproduction steps

1. Create a file structure with multiple nested folders to represent file shares, e.g.: /media/foo/TV Shows/  /media/bar/TV Shows/
2. Create a new Shows library in Jellyfin and add the TV Shows folders to the library.
3. Let the library index, and a random unrelated show from TMDB will be chosen and assigned to all content.

### What is the current _bug_ behavior?

A random unrelated show is chosen from TMDB that does not match the content. Multiple instances of this show are placed in the library

### What is the expected _correct_ behavior?

If network mounts are all grouped in one folder, Jellyfin should walk the paths from the relative folder specified in the library path and correctly locate metadata for the shows.

It appears to be walking from the root and getting confused when there is more than one folder between / and where the shows are located.

### Jellyfin Server version

10.10.0+

### Specify version number

10.10.3

### Specify the build version

10.10.3

### Environment

```markdown
- OS: Debian 12
- Linux Kernel: 6.11.5+bpo-amd64
- Virtualization: Docker
- Networking: bridge
- Storage: SMB, shares mounted on host, with path mapped into Docker
```

### Jellyfin logs

```shell
[2024-12-29 06:40:07.120 +00:00] [INF] [6] Main: Jellyfin version: "10.10.3"
[2024-12-29 06:40:07.136 +00:00] [INF] [6] Main: Environment Variables: ["[JELLYFIN_DATA_DIR, /config]", "[JELLYFIN_FFMPEG, /usr/lib/jellyfin-ffmpeg/ffmpeg]", "[JELLYFIN_LOG_DIR, /config/log]", "[JELLYFIN_CONFIG_DIR, /config/config]", "[JELLYFIN_WEB_DIR, /jellyfin/jellyfin-web]", "[JELLYFIN_CACHE_DIR, /cache]"]
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: Arguments: ["/jellyfin/jellyfin.dll"]
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: Operating system: "Debian GNU/Linux 12 (bookworm)"
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: Architecture: X64
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: 64-Bit Process: True
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: User Interactive: True
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: Processor count: 4
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: Program data path: "/config"
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: Log directory path: "/config/log"
[2024-12-29 06:40:07.138 +00:00] [INF] [6] Main: Config directory path: "/config/config"
[2024-12-29 06:40:07.139 +00:00] [INF] [6] Main: Cache path: "/cache"
[2024-12-29 06:40:07.139 +00:00] [INF] [6] Main: Temp directory path: "/tmp/jellyfin"
[2024-12-29 06:40:07.139 +00:00] [INF] [6] Main: Web resources path: "/jellyfin/jellyfin-web"
[2024-12-29 06:40:07.139 +00:00] [INF] [6] Main: Application directory: "/jellyfin/"

[2024-12-29 06:46:13.738 +00:00] [INF] [23] MediaBrowser.Providers.TV.SeriesMetadataService: Creating Season "Season Unknown" entry for "Midnight Family"

[2024-12-29 06:59:13.595 +00:00] [INF] [24] MediaBrowser.Providers.TV.SeriesMetadataService: Creating Season "Season 1" entry for "Midnight Family"
[2024-12-29 06:59:14.741 +00:00] [INF] [19] MediaBrowser.Providers.TV.SeriesMetadataService: Creating Season "Season 2" entry for "Midnight Family"
[2024-12-29 06:59:14.914 +00:00] [INF] [22] MediaBrowser.Providers.TV.SeriesMetadataService: Creating Season "Season 3" entry for "Midnight Family"
```

### FFmpeg logs

```shell

```

**Comment by felix920506** [comment]
The correct way would be to add /media/share1/tvshows, /media/share2/tvshows ... /media/shareX/tvshows all as paths in the library. Please head to our forum or chat rooms for troubleshooting if you have further questions / issues. https://jellyfin.org/contact

**Comment by flux-inductor** [comment]
> The correct way would be to add /media/share1/tvshows, /media/share2/tvshows ... /media/shareX/tvshows all as paths in the library.

Per the details I provided, doing exactly this is what triggers the issue.

I did not come to ask for help as I have already resolved the issue with the workaround I documented above. Rather, I feel this is a defect with the metadata handler's path parsing.

Only when removing one level of the path - i.e. truncating it to /media1/tvshows - does it work correctly. Jellyfin seems unable to handle more than one folder nesting above the root for TV Show media.
